### ISSUE TITLE
Linker fails with APPCENTER_DONT_USE_NATIVE_STARTER switch

### ISSUE THREAD

**Issue by minorai** [Microsoft/AppCenter-SDK-Unity#444]
### **Description**

Xcode build fails with APPCENTER_DONT_USE_NATIVE_STARTER switch

### **Repro Steps**

1. Create a new empty unity project
2. Add AppCenter Unity SDK (either by AppCenterEditorExtensions or unitypackages)
3. Add to defines APPCENTER_DONT_USE_NATIVE_STARTER switch in Player Settings
4. Build iOS Xcode project
5. Try to build and deploy Xcode project

Build fails because of the missing dependencies.

### **Details**
- AppCenterUnity SDK 3.1.0
- Unity 2018.4.20f1 LTS

The problem is because APPCENTER_DONT_USE_NATIVE_STARTER disables generation of AppCenterStarter.m module. A workaround and probably the solution is to add 

`@import AppCenter;`

into stub file. This will eliminate missing linker error.

P.S: on the side note. If the project ever did run a a build without APPCENTER_DONT_USE_NATIVE_STARTER it will not help to build it again with APPCENTER_DONT_USE_NATIVE_STARTER flag, as the AppCenterStarter.m file is already overriden. But that is a separate issue.

**Comment by minorai** [comment]
![image](https://user-images.githubusercontent.com/16007058/78661974-ea23ce80-78cf-11ea-8923-18ff4a26376f.png)
Missing symbols, just for clarity

**Comment by olmisa** [comment]
Hi @minorai ,
I was not able to reproduce this issue. Maybe it is possible to share a sample project with a reproduction code?
Also, could you try to select 'CoreTelephony' in 'AppCenter.framework' dependencies? I mean 'Import settings' in Unity. 

Thanks

**Comment by minorai** [comment]
I've tried to add CoreTelephony before. It takes care only of one missing symbol: "_OBJC_CLASS_$_CTTelephonyNetworkInfo", the rest seem to come from some compression library that is linked on demand, that is not part of any framework (at least I tried to include all of them during testing and it did not work).

The clue whenever the bug will trigger is the content of AppCenterStarter.m file, if it looks like this:
```
@implementation AppCenterStarter
//This file is a stub! We're going to replace it at build.
@end
```

You can find the project here, note that I've only compiled for devices (not a simulator).
https://send.firefox.com/download/77b8632c2135f00f/#13xJJSLajj3yGmP9zZh5Cg

**Comment by MatkovIvan** [comment]
@minorai thanks for sharing repro sample. We're looking at it now.

> the rest seem to come from some compression library

Yes, this comes form zlib (this is system library, not Objective-C framework). It can be temporary added directly by adding linker flag in XCode: `-lz`.

**Comment by minorai** [comment]
Thank you @MatkovIvan, I suspected as much. However I like my workaround better. Just by adding `@import AppCenter` in any .m file resolves all linking issues.

**Comment by AnastasiaKubova** [comment]
The fix had been released in [3.2.0](https://github.com/microsoft/appcenter-sdk-unity/releases/tag/3.2.0).
